common/djangoapps/student/models.py

```diff
@@ -1591,14 +1744,55 @@ def emit_event(self, event_name, enterprise_uuid=None):
             segment_traits['email'] = self.user.email
 
             if event_name == EVENT_NAME_ENROLLMENT_ACTIVATED:
+                request = crum.get_current_request()
+                user_timezone_locale = user_timezone_locale_prefs(request)
+                user_timezone = timezone(user_timezone_locale['user_timezone'] or str(UTC))
+
+                course = get_course_with_access(self.user, 'load', self.course_id, check_if_enrolled=False)
+                blocks = get_course_date_blocks(course, self.user, request, include_access=True,
+                                                include_past_dates=True)
+                blocks = [block for block in blocks if not isinstance(block, TodaysDate)]
+                date_serializer = DateSummarySerializer(blocks, many=True)
+                course_date_list = self._get_course_dates_for_email(date_serializer.data, user_timezone)
```

Comment:
can we encapsulate all these course_dates preparation lines into a function and only get final course_dates here? because we are only interested in getting the dates here?